# Awesome Genome Accelerator
A Collection of Papers on Genome Accelerators at top-tier conferences (e.g., HPCA, ASPLOS, ISCA, MICRO, ISSCC, DAC).

---

| Year | Conf.      | Paper                                                                                                                                                        | Affiliation |
| ---- | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ | ---- |
| 2018 | HPCA       | Searching for Potential gRNA Off-Target Sites for CRISPR/Cas9 Using Automata Processing Across Different Platforms                                           | UVA  |
| 2018 | ASPLOS     | Darwin: A Genomics Co-processor Provides up to 15000X Acceleration on Long Read Assembly                                                                     | Stanford University |
| 2018 | ISCA       | GenAx: A Genome Sequencing Accelerator                                                                                                                       | UofM |
||||<br>
| 2019 | HPCA       | Darwin-WGA: A Co-processor Provides Increased Sensitivity in Whole Genome Alignments with High Speed                                                         | Stanford University |
| 2019 | HPCA       | FPGA Accelerated INDEL Realignment in the Cloud                                                                                                              | UCB  |
| 2019 | MICRO      | GenCache: Leveraging In-Cache Operators for Efficient Sequence Alignment                                                                                     | University of Utah |
| 2019 | MICRO      | MEDAL: Scalable DIMM-based Near Data Processing Accelerator for DNA Seeding Algorithm                                                                        | UCSB  |
||||<br>
| 2020 | ISCA       | Genesis: A Hardware Acceleration Framework for Genomic Data Analysis                                                                                         | SNU |
| 2020 | MICRO      | SeedEx: A Genome Sequencing Accelerator for Optimal Alignments in Subminimal Space                                                                           | UofM  |
| 2020 | MICRO      | GenASM: A High-Performance, Low-Power Approximate String Matching Acceleration Framework for Genome Sequence Analysis                                        | ETHZ |
| 2020 | ISSCC      | A Fully Integrated Genetic Variant Discovery SoC for Next-Generation Sequencing                                                                              | NTU  |
| 2020 | DAC        | PIM-Assembler: A Processing-in-Memory Platform for Genome Assembly                                                                                           | Arizona State University  |
||||<br>
| 2021 | HPCA       | EXMA: A Genomics Accelerator for Exact-Matching                                                                                                              | Indiana University Bloomington |
| 2021 | ISCA       | Accelerated Seeding for Genome Sequence Alignment with Enumerated Radix Trees                                                                                | UofM  |
| 2021 | ISCA       | Sieve: Scalable In-situ DRAM-based Accelerator Designs for Massively Parallel k-mer Matching                                                                 | UVA |
| 2021 | MICRO      | SquiggleFilter: An Accelerator for Portable Virus Detection                                                                                                  | UofM  |
||||<br>
| 2022 | ASPLOS     | GenStore: a high-performance in-storage processing system for genome sequence analysis                                                                       | ETHZ |
| 2022 | ASPLOS     | ProSE: the architecture and design of a protein discovery engine                                                                                             | Duke  |
| 2022 | ISCA       | SeGraM: a universal hardware accelerator for genomic sequence-to-graph and sequence-to-sequence mapping                                                      | ETHZ |
| 2022 | ISCA       | BioHD: an efficient genome sequence search platform using HyperDimensional memorization                                                                      | UCI  |
| 2022 | ISCA       | EDAM: Edit Distance tolerant Approximate Matching content addressable memory                                                                                 | Technion-Israel Institute of Technology |
| 2022 | MICRO      | BEACON: Scalable Near-Data-Processing Accelerators for Genome Analysis near Memory Pool with the CXL Support                                                 | UCSB |
| 2022 | MICRO      | GenPIP: In-Memory Acceleration of Genome Analysis via Tight Integration of Basecalling and Read Mapping                                                      | ETHZ |
||||<br>
| 2023 | HPCA       | NvWa: Enhancing Sequence Alignment Accelerator Throughput via Hardware Scheduling                                                                            | ICT, CAS |
| 2023 | ISCA       | GenDP: A Framework of Dynamic Programming Acceleration for Genome Sequencing Analysis                                                                        | UofM  |
| 2023 | MICRO      | Swordfish: A Framework for Evaluating Deep Neural Network-based Basecalling using Computation-In-Memory with Non-Ideal Memristors                            | ETHZ |
| 2023 | MICRO      | DASH-CAM: Dynamic Approximate SearcH Content Addressable Memory for genome classification                                                                    | Bar-Ilan University  |
| 2023 | MICRO      | CASA: An Energy-Efficient and High-Speed CAM-based SMEM Seeding Accelerator for Genome Alignment                                                             | THU |
| 2023 | MICRO      | GMX: Instruction Set Extensions for Fast, Scalable, and Efficient Genome Sequence Alignment                                                                  | BSC |
| 2023 | ISSCC      | A Fully Integrated End-to-End Genome Analysis Accelerator for Next-Generation Sequencing                                                                     | NTU |
| 2023 | DAC        | Invited: Accelerating Genome Analysis via Algorithm-Architecture Co-Design                                                                                   | ETHZ |
| 2023 | DAC        | ASMCap: An Approximate String Matching Accelerator for Genome Sequence Analysis Based on Capacitive Content Addressable Memory                               | THU |
| 2023 | DAC        | MeG2: In-Memory Acceleration for Genome Graphs Analysis                                                                                                      | HUST |
||||<br>
| 2024 | HPCA       | TALCO: Tiling Genome Sequence Alignment Using Convergence of Traceback Pointers                                                                              | UCSD |
| 2024 | ASPLOS     | Harp: Leveraging Quasi-Sequential Characteristics to Accelerate Sequence-to-Graph Mapping of Long Reads                                                      | THU  |
| 2024 | ISCA       | QUETZAL: Vector Acceleration Framework For Modern Genome Sequence Analysis                                                                                   | BSC |
| 2024 | ISCA       | BLESS: Bandwidth and Locality Enhanced SMEM Seeding Acceleration for DNA Sequencing                                                                          | KAIST |
| 2024 | ISCA       | MegIS: High-Performance, Energy-Efficient, and Low-Cost Metagenomic Analysis with In-Storage Processing                                                      | ETHZ |




<!--

## HPCA 2018
Searching for Potential gRNA Off-Target Sites for CRISPR/Cas9 Using Automata Processing Across Different Platforms, UVA

## ASPOLS 2018
Darwin: A Genomics Co-processor Provides up to 15000X Acceleration on Long Read Assembly, Stanford University

## ISCA 2018
GenAx: A Genome Sequencing Accelerator, UofM

## MICRO 2018
NONE

## ISSCC 2018
NONE

## DAC 2018
NONE

---

## HPCA 2019
Darwin-WGA: A Co-processor Provides Increased Sensitivity in Whole Genome Alignments with High Speed, Stanford University

FPGA Accelerated INDEL Realignment in the Cloud, UCB

## ASPLOS 2019
NONE

## ISCA 2019
NONE

## MICRO 2019
GenCache: Leveraging In-Cache Operators for Efficient Sequence Alignment, University of Utah

MEDAL: Scalable DIMM-based Near Data Processing Accelerator for DNA Seeding Algorithm, UCSB

## ISSCC 2019
NONE

## DAC 2019
NONE

---



## HPCA 2020
NONE

## ASPLOS 2020
NONE

## ISCA 2020
Genesis: A Hardware Acceleration Framework for Genomic Data Analysis, SNU

## MICRO 2020
SeedEx: A Genome Sequencing Accelerator for Optimal Alignments in Subminimal Space, UofM

GenASM: A High-Performance, Low-Power Approximate String Matching Acceleration Framework for Genome Sequence Analysis, ETHZ

## ISSCC 2020
NONE

## DAC 2020
PIM-Assembler: A Processing-in-Memory Platform for Genome Assembly, Arizona State University

---

## HPCA 2021
EXMA: A Genomics Accelerator for Exact-Matching, Indiana University Bloomington

## ASPLOS 2021
NONE

## ISCA 2021
Accelerated Seeding for Genome Sequence Alignment with Enumerated Radix Trees, UofM

Sieve: Scalable In-situ DRAM-based Accelerator Designs for Massively Parallel k-mer Matching, UVA

## MICRO 2021
SquiggleFilter: An Accelerator for Portable Virus Detection, UofM

## ISSCC 2021
NONE

## DAC 2021
NONE

---

## HPCA 2022
NONE

## ASPLOS 2022
GenStore: a high-performance in-storage processing system for genome sequence analysis, ETHZ

ProSE: the architecture and design of a protein discovery engine, Duke

## ISCA 2022
SeGraM: a universal hardware accelerator for genomic sequence-to-graph and sequence-to-sequence mapping, ETHZ

BioHD: an efficient genome sequence search platform using HyperDimensional memorization, UCI

EDAM: Edit Distance tolerant Approximate Matching content addressable memory, Technion-Israel Institute of Technology

## MICRO 2022
BEACON: Scalable Near-Data-Processing Accelerators for Genome Analysis near Memory Pool with the CXL Support, UCSB

GenPIP: In-Memory Acceleration of Genome Analysis via Tight Integration of Basecalling and Read Mapping, ETHZ

## ISSCC 2022
NONE

## DAC 2022
NONE

---

## HPCA 2023
NvWa: Enhancing Sequence Alignment Accelerator Throughput via Hardware Scheduling, ICT,CAS

## ASPLOS 2023
NONE

## ISCA 2023
GenDP: A Framework of Dynamic Programming Acceleration for Genome Sequencing Analysis, UofM

## MICRO 2023
Swordfish: A Framework for Evaluating Deep Neural Network-based Basecalling using Computation-In-Memory with Non-Ideal Memristors, ETHZ

DASH-CAM: Dynamic Approximate SearcH Content Addressable Memory for genome classification, Bar-Ilan University

CASA: An Energy-Efficient and High-Speed CAM-based SMEM Seeding Accelerator for Genome Alignment, THU

GMX: Instruction Set Extensions for Fast, Scalable, and Efficient Genome Sequence Alignment, Barcelona Supercomputing Center

## ISSCC 2023
A Fully Integrated End-to-End Genome Analysis Accelerator for Next-Generation Sequencing, NTU

## DAC 2023
Invited: Accelerating Genome Analysis via Algorithm-Architecture Co-Design, ETHZ

ASMCap: An Approximate String Matching Accelerator for Genome Sequence Analysis Based on Capacitive Content Addressable Memory, THU

MeG2: In-Memory Acceleration for Genome Graphs Analysis, HUST

---

## HPCA 2024
TALCO: Tiling Genome Sequence Alignment Using Convergence of Traceback Pointers, UCSD

## ASPLOS 2024
Harp: Leveraging Quasi-Sequential Characteristics to Accelerate Sequence-to-Graph Mapping of Long Reads, THU

## ISCA 2024
QUETZAL: Vector Acceleration Framework For Modern Genome Sequence Analysis, Barcelona Supercomputing Center

BLESS: Bandwidth and Locality Enhanced SMEM Seeding Acceleration for DNA Sequencing, KAIST

MegIS: High-Performance, Energy-Efficient, and Low-Cost Metagenomic Analysis with In-Storage Processing, ETHZ

## MICRO 2024
TBD...

## ISSCC 2024
NONE

## DAC 2024
TBD...
-->

